Source/PersonalityInsightsV3/PersonalityInsights.swift

```diff
@@ -0,0 +1,274 @@
+            rawScores: rawScores,
+            consumptionPreferences: consumptionPreferences,
+            failure: failure,
+            success: success
+        )
+    }
+
+    /**
+     Analyze content to generate a personality profile.
+ 
+     - parameter version: The date in the form "YYYY-MM-DD" that will specify a version no later than the date provided.
+     - parameter content: The content to analyze.
+     - parameter contentType: The MIME content-type of the content.
+     - parameter acceptLanguage: The desired language of the response.
+     - parameter contentLanguage: The language of the text being analyzed.
+     - parameter rawScores: If true, then a raw score for each characteristic is returned in
+        addition to a normalized score. Raw scores are not compared with a sample population.
+        A raw sampling error for each characteristic is also returned.
+     - parameter consumptionPreferences: If true, then information inferred about consumption preferences is returned in addition to the results.
+     - parameter failure: A function executed if an error occurs.
+     - parameter success: A function executed with the personality profile.
+     */
+    private func getProfile(
+        version: String,
+        fromContent content: Data?,
+        withType contentType: String,
+        acceptLanguage: String? = nil,
+        contentLanguage: String? = nil,
+        rawScores: Bool? = nil,
+        consumptionPreferences: Bool? = nil,
+        failure: ((Error) -> Void)? = nil,
+        success: @escaping (Profile) -> Void)
+    {
+        // construct query parameters
+        var queryParameters = [URLQueryItem]()
+        if let rawScores = rawScores {
+            let queryParameter = URLQueryItem(name: "raw_scores", value: "\(rawScores)")
+            queryParameters.append(queryParameter)
+        }
+        if let consumptionPreferences = consumptionPreferences {
+            let consumptionPreferenceQueryParameter = URLQueryItem(name: "consumption_preferences", value: "\(consumptionPreferences)")
+            queryParameters.append(consumptionPreferenceQueryParameter)
+        }
+        let versionQueryParameter = URLQueryItem(name: "version", value: version)
+        queryParameters.append(versionQueryParameter)
+
+        // construct header parameters
+        var headerParameters = defaultHeaders
+        if let acceptLanguage = acceptLanguage {
+            headerParameters["Accept-Language"] = acceptLanguage
+        }
+        if let contentLanguage = contentLanguage {
+            headerParameters["Content-Language"] = contentLanguage
+        }
+
+        // construct REST request
+        let request = RestRequest(
+            method: "POST",
+            url: serviceURL + "/v3/profile",
+            credentials: credentials,
+            headerParameters: headerParameters,
+            acceptType: "application/json",
+            contentType: contentType,
```

Comment:
It definitely took me awhile to understand where the `acceptType` and `contentType` was being defined, so I actually would prefer removing those parameters from the initializer. When I was using Personality Insight's API reference I kept expecting to find those parameters in the function definition of `getProfile`so it took a little longer finding out that they were within RestRequest's initializer.